torch/fx/experimental/Partitioner.py

```diff
@@ -1,7 +1,92 @@
 from torch.fx.graph_module import GraphModule
 from torch.fx.node import Node
-from torch.fx.experimental import GraphManipulation
-from typing import Dict, List, Union
+from typing import Dict, List, Union, Tuple
+from torch.fx.experimental.GraphManipulation import get_all_users_of
+def _get_users_for_all_nodes(fx_module: GraphModule) -> Dict[int, List[int]]:
+    """Given a fx module, return a dictionary with the index of a node as a key and
+       a list of its user nodes' indexes as value.
+    """
+    users_mapping: Dict[int, List[int]] = {}
+    for i, node in enumerate(fx_module.graph.nodes):
+        if node.op == 'output':
+            continue
+        # Not conside output node
+        users: List[int] = get_all_users_of(fx_module, i)
+        # 'output' node is not considered.
+        output_node_index = len(fx_module.graph.nodes) - 1
+        if output_node_index in users:
+            users.remove(output_node_index)
+        users_mapping[i] = users
+    return users_mapping
+
+def _get_args_for_all_nodes(users_mapping: Dict[int, List[int]]) -> Dict[int, List[int]]:
+    """Given a node-user mapping (index of node as key, its user index list as value),
+       return a dictionary with the index of a node as a key and
+       its arg node's index as value
+    """
+    args_mapping: Dict[int, List[int]] = {k: [] for k in users_mapping}
+    for k, v in users_mapping.items():
+        for user in v:
+            if k not in args_mapping[user]:
+                args_mapping[user].append(k)
+    return args_mapping
+
+def _get_bfs_levels(fx_module: GraphModule) -> Dict[str, object]:
+    """Given a fx module, return a list of node indexes in a bfs level order
+       for example, a graph like this (numbers are the indexes of nodes):
```

Comment:
Node indexes are not guaranteed to be stable. In fact, after https://github.com/pytorch/pytorch/pull/45708, Nodes will not be addressable by index. We should sync up here